### DISCUSSION TITLE
Code splitting within single file

### DISCUSSION THREAD

**Discussion by zdila** [webpack/webpack#18428] [Ideas]
Hello. I wish there was a support for code splitting within single file, for example with [module expression](https://github.com/tc39/proposal-module-expressions) syntax :

```js
const mod = module {
  export const foo = 1;
};

const bar = await import(mod);

console.log(bar.foo);
```

Webpack would produce 2 files:

`file1.js`:
```
export const foo = 1;
```

`file2.js`:
```
const bar = await import('file1.js');

console.log(bar.foo);
```

In our app we have many modules that mostly only add some condition about when to lazy-load other module and the dynamic import itself. It would make our code less cluttered if both parts would be in a single file as they belong together. [Example](https://github.com/FreemapSlovakia/freemap-v3-react/blob/develop/src/processors/galleryItemUploadProcessor.ts).

I am not sure if it could be part of dedicated Babel plugin, part of a Webpack or both. What do you think?

**Comment by zdila** [comment]
Actually I've added a proposal about this long time ago and forgot about it - https://github.com/tc39/proposal-module-expressions/issues/52

**Reply by alexander-akait** [reply]
It should be part of webpack too, but it is still `Stage 3`, we try to implement only stage 4, because sometimes things change, like it was with import assertion
